### DISCUSSION TITLE
What is the best way to programmatically change a field value?

### DISCUSSION THREAD

**Discussion by samuel-morrissey** [TanStack/form#728] [General] [answered]
I have some use cases when I want to get the value of the field and others when I want to programmatic change its value. I tried using useStore, useField,  getFieldValue, and setFieldValue but all this seems redundant and not straightforward. Sometimes I have weird behaviors,  and sometimes I need to put work around props (like explicit put touch: true).
So, there is any right way to do this? What the intention of expose all those apis? Can we discuss a simple way to do that?

**Comment by crutchcorn** [comment, marked as answer]
It's super redundant right now and we need clearer guidance in our docs and beyond.

Use `setFieldValue` with `{touch: true}`. In the future we'll remove the `touch` API and _maybe_ even mark `getFieldValue` as `@deprecated` to stop folks outside of our internal code from using it.

**Reply by Jee-vim** [reply]
> It's super redundant right now and we need clearer guidance in our docs and beyond.
> 
> Use `setFieldValue` with `{touch: true}`. In the future we'll remove the `touch` API and _maybe_ even mark `getFieldValue` as `@deprecated` to stop folks outside of our internal code from using it.

any update on this? cause i try use `{touch: true}` is not work anymore
